addons/rose/addon/components/rose/code-editor/index.stories.mdx

```diff
@@ -0,0 +1,90 @@
+import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
+import { hbs } from 'ember-cli-htmlbars';
+
+<Meta title="Rose|Code Editor" component="CodeEditor" />
+
+# Code Editor
+
+`CodeEditor` is a component that yields two presentational components which together provide an IDE-like UX.
+CodeEditor::FieldEditor is where the provided code is displayed with syntax highlighting.
+CodeEditor::Toolbar is intended to go on top of the FieldEditor and provides a 'Copy code' button as well as other optional user-provided actions.
+
+```html
+<CodeEditor @codeValue="{{this.codeIWantToSee}}" as |C|>
+  <C.Toolbar />
+  <C.FieldEditor />
+</CodeEditor>
+```
+
+<!-- mdx seems to not be able to handle template literals -->
+
+export const codeValue =
+  'hcp_boundary_cluster_id = "<cluster-id>"\n\n' +
+  'listener "tcp" {\n' +
+  '   address = "0.0.0.0:9202"\n' +
+  '   purpose = "proxy"\n' +
+  '}\n\n' +
+  'worker {\n' +
+  '   public_addr = "107.22.128.152"\n' +
+  '   auth_storage_path = "/home/ubuntu/boundary/worker1"\n' +
+  '   tags {\n' +
+  '      type = ["dev-worker", "ubuntu"]\n' +
+  '   }\n' +
+  '}';
+
+<Preview>
+  <Story name="Readonly">
+    {{
+      template: hbs`
+      <Rose::CodeEditor @codeValue={{codeValue}} as |C|>
+        <C.Toolbar />
+        <C.FieldEditor @options={{options}} />
+      </Rose::CodeEditor>`,
+      context: {
+        codeValue: codeValue,
+        options: {
+          mode: 'hcl',
+          readOnly: true,
+        },
+      },
+    }}
+  </Story>
+</Preview>
+
+<Preview>
+  <Story name="CodeEditorShell">
+    {{
+      template: hbs`
+      <Rose::CodeEditor @codeValue={{codeValue}} as |C|>
+        <C.Toolbar />
+        <C.FieldEditor class='autosize' @options={{options}} />
```

Comment:
I think we'd want to make "autosize" an argument, rather than passing a classname like this.

Reply:
I made it an argument but can you give a little explanation of what's the difference or why it's better?